mp/src/game/server/momentum/mom_triggers.cpp

```diff
@@ -1560,26 +1561,38 @@ void CNoGrenadesZone::Precache()
 
 bool CNoGrenadesZone::IsInsideNoGrenadesZone(CBaseEntity *pOther)
 {
-    if ( pOther )
+    if (!pOther)
+        return false;
+
+    auto pSticky = dynamic_cast<CMomStickybomb *>(pOther);
+    FOR_EACH_VEC(s_vecNoGrenadeZones, i)
     {
-        const auto pSticky = dynamic_cast<CMomStickybomb*>(pOther);
-        FOR_EACH_VEC(s_vecNoGrenadeZones, i)
-        {
-            const auto pNoGrenadeZone = s_vecNoGrenadeZones[i];
+        const auto pNoGrenadeZone = s_vecNoGrenadeZones[i];
 
-            if (pNoGrenadeZone->m_bDisabled || !pNoGrenadeZone->PointIsWithin(pOther->GetAbsOrigin()))
-                continue;
+        if (pNoGrenadeZone->m_bDisabled || !pNoGrenadeZone->PointIsWithin(pOther->GetAbsOrigin()))
+            continue;
 
-            if (!pSticky)
-                return true; // This is a rocket in a nonade zone and should fizzle
+        if (!pSticky)
+            return true; // This is a rocket in a nonade zone and should fizzle
 
-            return !pNoGrenadeZone->m_bAirborneOnly || !pSticky->DidHitWorld();
-        }
+        return pNoGrenadeZone->m_iStickyFizzleType == FIZZLE_ON_LAND && !pNoGrenadeZone->m_bAirborneOnly || pSticky->DidHitWorld();
```

Comment:
Okay this is slightly confusing. This is a method to determine if the bomb itself is inside a nogrenades. This shouldn't be handling explosive logic, that's more for inside the stickybomb class.

I'd make this return the `CNoGrenadesZone` trigger that the given explosive was checked against if it found one, otherwise null. This will let you pull this kinda stuff out into the stickybomb class to do this extra logic checking.

Reply:
This function also handles logic for rockets; acting like `FIZZLE_ON_ENTRANCE` for them; should I make a new function entirely for just stickybomb checking or something?